### PR TITLE
COSI-41: expand-e2e-tests-for-bucket-access

### PR DESCRIPTION
**Pull request by anurag4DSB**
- Add e2e tests for bucket access.
- Verification for creation of IAM user, with inline policy, AK/SK, and secret used by applications to access object storage information over kube

### CODE REVIEW

**Comment by anurag4DSB** [comment]
I remembered the 3rd commit change after asking for reviews. I added a fixup commit in case someone has already begun reviewing it.

**Review by tmacro — APPROVED** [review]

File: .github/scripts/e2e_tests.sh [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+log_and_run echo "Verifying IAM user with prefix '$IAM_USER_PREFIX'..."
+log_and_run aws --endpoint-url "$IAM_ENDPOINT" iam list-users
+
+IAM_USER_NAME=$(aws --endpoint-url "$IAM_ENDPOINT" iam list-users --query "Users[?starts_with(UserName, '$IAM_USER_PREFIX')].UserName" --output text)
+if [ -z "$IAM_USER_NAME" ]; then
+  log_and_run echo "IAM user with prefix '$IAM_USER_PREFIX' not found."
+  exit 1
+else
+  log_and_run echo "IAM user found: $IAM_USER_NAME"
+fi
```
Comment (jonathan-gramain) [inline .github/scripts/e2e_tests.sh:134]:
```suggestion
fi
log_and_run echo "IAM user found: $IAM_USER_NAME"
```

File: .github/scripts/e2e_tests.sh [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+log_and_run echo "Verifying inline policy attached to IAM user..."
+INLINE_POLICY=$(aws --endpoint-url "$IAM_ENDPOINT" iam list-user-policies --user-name "$IAM_USER_NAME" --query "PolicyNames[0]" --output text)
+EXPECTED_INLINE_POLICY="$BUCKET_FOUND-cosi-ba"
+
+if [[ "$INLINE_POLICY" == "$EXPECTED_INLINE_POLICY" ]]; then
+  log_and_run echo "Inline policy '$INLINE_POLICY' matches bucket name '$EXPECTED_INLINE_POLICY'."
+else
+  log_and_run echo "Inline policy '$INLINE_POLICY' does not match expected bucket name '$EXPECTED_INLINE_POLICY'."
+  exit 1
+fi
```
Comment (jonathan-gramain) [inline .github/scripts/e2e_tests.sh:145]:
```suggestion
if [[ "$INLINE_POLICY" != "$EXPECTED_INLINE_POLICY" ]]; then
  log_and_run echo "Inline policy '$INLINE_POLICY' does not match expected bucket name '$EXPECTED_INLINE_POLICY'."
  exit 1
fi
log_and_run echo "Inline policy '$INLINE_POLICY' matches bucket name '$EXPECTED_INLINE_POLICY'."
```

File: .github/scripts/e2e_tests.sh [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+log_and_run echo "Verifying access key for IAM user..."
+IAM_USER_ACCESS_KEY=$(aws --endpoint-url "$IAM_ENDPOINT" iam list-access-keys --user-name "$IAM_USER_NAME" --query "AccessKeyMetadata[0].AccessKeyId" --output text)
+
+if [ -z "$IAM_USER_ACCESS_KEY" ]; then
+  log_and_run echo "Access key not found for IAM user '$IAM_USER_NAME'."
+  exit 1
+else
+  log_and_run echo "Access key found for IAM user: $IAM_USER_ACCESS_KEY"
+fi
```
Comment (jonathan-gramain) [inline .github/scripts/e2e_tests.sh:155]:
```suggestion
fi
log_and_run echo "Access key found for IAM user: $IAM_USER_ACCESS_KEY"
```

File: .github/scripts/e2e_tests.sh [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+# Step 10: Verify the object-storage-access-secret
+log_and_run echo "Verifying object-storage-access-secret in the default namespace..."
+
+# Fetch the secret data as JSON
+SECRET_JSON=$(kubectl get secret "$SECRET_NAME" --namespace "$NAMESPACE" -o json)
+
+# Decode the Base64 encoded BucketInfo
+BUCKET_INFO_BASE64=$(echo "$SECRET_JSON" | jq -r '.data.BucketInfo')
+BUCKET_INFO_JSON=$(echo "$BUCKET_INFO_BASE64" | base64 --decode)
```
Comment (jonathan-gramain) [inline .github/scripts/e2e_tests.sh:165]:
```suggestion
BUCKET_INFO_JSON="$(echo "$BUCKET_INFO_BASE64" | base64 --decode)"
```

File: .github/scripts/e2e_tests.sh [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+  exit 1
+else
+  log_and_run echo "Access key found for IAM user: $IAM_USER_ACCESS_KEY"
+fi
+
+# Step 10: Verify the object-storage-access-secret
+log_and_run echo "Verifying object-storage-access-secret in the default namespace..."
+
+# Fetch the secret data as JSON
+SECRET_JSON=$(kubectl get secret "$SECRET_NAME" --namespace "$NAMESPACE" -o json)
```
Comment (jonathan-gramain) [inline .github/scripts/e2e_tests.sh:161]:
```suggestion
SECRET_JSON="$(kubectl get secret "$SECRET_NAME" --namespace "$NAMESPACE" -o json)"
```

File: .github/scripts/e2e_tests.sh [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+IAM_USER_NAME=$(aws --endpoint-url "$IAM_ENDPOINT" iam list-users --query "Users[?starts_with(UserName, '$IAM_USER_PREFIX')].UserName" --output text)
+if [ -z "$IAM_USER_NAME" ]; then
+  log_and_run echo "IAM user with prefix '$IAM_USER_PREFIX' not found."
+  exit 1
+else
+  log_and_run echo "IAM user found: $IAM_USER_NAME"
+fi
+
+log_and_run echo "Verifying inline policy attached to IAM user..."
+INLINE_POLICY=$(aws --endpoint-url "$IAM_ENDPOINT" iam list-user-policies --user-name "$IAM_USER_NAME" --query "PolicyNames[0]" --output text)
```
Comment (jonathan-gramain) [inline .github/scripts/e2e_tests.sh:137]:
```suggestion
INLINE_POLICY="$(aws --endpoint-url "$IAM_ENDPOINT" iam list-user-policies --user-name "$IAM_USER_NAME" --query "PolicyNames[0]" --output text)"
```

**Review by jonathan-gramain — APPROVED** [review]
mostly nitpick suggestions, systematically quoting the `"$(...)"` can avoid some surprises if the results contain spaces for some reason.

**Pull request merged**